operator/cloudscale/provision.go

```diff
@@ -0,0 +1,92 @@
+package cloudscale
+
+import (
+	"context"
+	"fmt"
+	pipeline "github.com/ccremer/go-command-pipeline"
+	cloudscalev1 "github.com/vshn/appcat-service-s3/apis/cloudscale/v1"
+	"github.com/vshn/appcat-service-s3/apis/conditions"
+	"github.com/vshn/appcat-service-s3/operator/steps"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+// ObjectsUserPipeline provisions ObjectsUsers on cloudscale.ch
+type ObjectsUserPipeline struct {
+}
+
+// NewObjectsUserPipeline returns a new instance of ObjectsUserPipeline.
+func NewObjectsUserPipeline() *ObjectsUserPipeline {
+	return &ObjectsUserPipeline{}
+}
+
+// Run executes the business logic.
+func (p *ObjectsUserPipeline) Run(ctx context.Context) error {
+	pipe := pipeline.NewPipeline().WithBeforeHooks(debugLogger(ctx)).
+		WithSteps(
+			pipeline.NewStepFromFunc("add finalizer", steps.AddFinalizerFn(ObjectsUserKey{}, userFinalizer)),
+			pipeline.NewStepFromFunc("create client", CreateCloudscaleClientFn(APIToken)),
+			pipeline.IfOrElse(isObjectsUserIDKnown(),
+				pipeline.NewStepFromFunc("fetch objects user", GetObjectsUserFn()),
```

Comment:
I get that it's convenient to use functions that return functions when there's a need to pass arguments to steps. But for normal steps that don't require arguments this makes it more complicated than necessary. Is there a specific reason you wrapped it this way?

Reply:
It's just for consistency, to follow the same scheme.
Would you prefer they can be used as funcs directly if there's no arg?

Reply:
For the sake of simplicity yes. It could be confusing for others as well.

I'd favor simplicity over consistency here.